packages/teams-js/src/internal/nestedAppAuth.ts

```diff
@@ -0,0 +1,209 @@
+  handlePostMessage: (message: string) => void;
+};
+
+/**
+ * @hidden
+ * Attempt to polyfill the nestedAppAuthBridge object on the given window
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function tryPolyfillWithNestedAppAuthBridge(
+  clientSupportedSDKVersion: string,
+  window: Window | null,
+  handlers: NestedAppAuthBridgeHandlers,
+): void {
+  const logger = tryPolyfillWithNestedAppAuthBridgeLogger;
+
+  if (GlobalVars.isFramelessWindow) {
+    logger('Cannot polyfill nestedAppAuthBridge as current window is frameless');
+    return;
+  }
+
+  if (!window) {
+    logger('Cannot polyfill nestedAppAuthBridge as current window does not exist');
+    return;
+  }
+
+  const parsedClientSupportedSDKVersion = (() => {
+    try {
+      return JSON.parse(clientSupportedSDKVersion);
+    } catch (e) {
+      return null;
+    }
+  })();
+
+  if (!parsedClientSupportedSDKVersion || !parsedClientSupportedSDKVersion.supports?.nestedAppAuth) {
+    logger('Cannot polyfill nestedAppAuthBridge as current hub does not support nested app auth');
+    return;
+  }
+
+  const extendedWindow = window as unknown as NestedAuthExtendedWindow;
+  if (extendedWindow.nestedAppAuthBridge) {
+    logger('nestedAppAuthBridge already exists on current window, skipping polyfill');
+    return;
+  }
+
+  const nestedAppAuthBridge = createNestedAppAuthBridge(extendedWindow, handlers);
+  if (nestedAppAuthBridge) {
+    extendedWindow.nestedAppAuthBridge = nestedAppAuthBridge;
+  }
+}
+
+const createNestedAppAuthBridgeLogger = nestedAppAuthLogger.extend('createNestedAppAuthBridge');
+
+/**
+ * @hidden
+ * Creates a bridge for nested app authentication.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @param {Window | null} window - The window object where the nested app authentication bridge will be created. If null, the function will log an error message and return null.
+ * @returns {NestedAppAuthBridge | null} Returns an object with methods for adding and removing event listeners, and posting messages. If the provided window is null, returns null.
+ *
+ * @property {Function} addEventListener - Adds an event listener to the window. Only supports the 'message' event. If an unsupported event is passed, logs an error message.
+ * @property {Function} postMessage - Posts a message to the window. The message should be a stringified JSON object with a messageType of 'NestedAppAuthRequest'. If the message does not meet these criteria, logs an error message.
+ * @property {Function} removeEventListener - Removes an event listener from the window.
+ */
+function createNestedAppAuthBridge(
+  window: Window | null,
+  bridgeHandlers: NestedAppAuthBridgeHandlers,
+): NestedAppAuthBridge | null {
+  const logger = createNestedAppAuthBridgeLogger;
+
+  if (!window) {
+    logger('nestedAppAuthBridge cannot be created as current window does not exist');
+    return null;
+  }
+
+  const { onMessage, handlePostMessage } = bridgeHandlers;
```

Comment:
very minor since this is internal only, but the name `handlePostMessage` makes me thing that this function is receiving and processing postMessages, when it's actually sending them. Maybe a rename to something like `sendPostMessage`?